src/api/swarm_api.go

```diff
@@ -0,0 +1,137 @@
+			TextLogging:    false,
+			JournalLogging: true,
+		},
+		DryRunConfig: agentCfg.DryRunConfig,
+	}
+	installerCfg := &installerConfig.Config{
+		DryRunConfig: installerConfig.DryRunConfig{
+			DryRunEnabled:          true,
+			FakeRebootMarkerPath:   params.NewAgentParams.DryFakeRebootMarkerPath,
+			ForcedHostID:           params.NewAgentParams.DryForcedHostID.String(),
+			DryRunClusterHostsPath: params.NewAgentParams.DryClusterHostsPath,
+		},
+		InfraEnvID:           params.NewAgentParams.InfraEnvID.String(),
+		URL:                  params.NewAgentParams.ServiceURL,
+		AgentImage:           params.NewAgentParams.AgentVersion,
+		PullSecretToken:      params.NewAgentParams.PullSecret,
+		SkipCertVerification: true,
+		CACertPath:           params.NewAgentParams.Cacert,
+	}
+	logSenderCfg := &agentConfig.LogsSenderConfig{
+		AgentConfig:     *agentCfg,
+		InfraEnvID:      params.NewAgentParams.InfraEnvID.String(),
+		TargetURL:       params.NewAgentParams.ServiceURL,
+		PullSecretToken: params.NewAgentParams.PullSecret,
+	}
+	log := util.NewJournalLogger("combined-agent", params.NewAgentParams.DryForcedHostID.String())
+	if err := installerConfig.DryParseClusterHosts(params.NewAgentParams.DryClusterHostsPath, &installerCfg.ParsedClusterHosts); err != nil {
+		log.WithError(err).Error("DryParseClusterHosts")
+		return swarm.NewCreateNewAgentBadRequest().WithPayload(newError(http.StatusBadRequest, err))
+	}
+	newAgent, err := agents.NewAgent(id)
+	if err != nil {
+		log.WithError(err).Warnf("Cannot allocate agent %d id %s", id, params.NewAgentParams.DryForcedHostID.String())
+		return swarm.NewDeleteAgentInternalServerError().WithPayload(newError(http.StatusInternalServerError, err))
+	}
+	go func() {
+		defer agents.SetTerminated(id)
+
+		factory := factories.NewNextStepRunnerFactory(subprocessCfg, installerCfg, logSenderCfg, log)
+		agent.RunAgent(agentCfg, factory, log)
+	}()
+	return swarm.NewCreateNewAgentCreated().WithPayload(newAgent)
+}
+
+func (s *SwarmAPI) DeleteAgent(ctx context.Context, params swarm.DeleteAgentParams) middleware.Responder {
+	err := agents.DeleteAgent(params.AgentID)
+	if err != nil {
+		return swarm.NewDeleteAgentNotFound().WithPayload(newError(http.StatusNotFound, err))
+	}
+	return swarm.NewDeleteAgentNoContent()
+}
+
+func (s *SwarmAPI) GetAgent(ctx context.Context, params swarm.GetAgentParams) middleware.Responder {
+	agent := agents.GetAgent(params.AgentID)
+	if agent == nil {
+		return swarm.NewGetAgentNotFound().WithPayload(newError(http.StatusNotFound, fmt.Errorf("id %d was not found", params.AgentID)))
+	}
+	return swarm.NewGetAgentOK().WithPayload(agent)
+}
+
+func (s *SwarmAPI) ListAgents(ctx context.Context, params swarm.ListAgentsParams) middleware.Responder {
+	return swarm.NewListAgentsOK().WithPayload(agents.ListAgents())
+}
+
+func (s *SwarmAPI) Exit(ctx context.Context, params swarm.ExitParams) middleware.Responder {
+	go func() {
+		time.Sleep(500 * time.Millisecond)
```

Comment:
What's the reason for this sleep?

Reply:
To let send the reply to the caller before termination

Reply:
Ah okay. Can you add that as a code comment?

Reply:
OK